scenes/game_logic/erratic_walk_behavior.gd

```diff
@@ -0,0 +1,73 @@
+# SPDX-FileCopyrightText: The Threadbare Authors
+# SPDX-License-Identifier: MPL-2.0
+@tool
+class_name ErraticWalkBehavior
+extends Node2D
+## Make the character walk around erratically.
+##
+## TODO:
+## - Emit signals and hook with ThrowingEnemy
+## - Try it in another game entity
+## - Constrain to an area
+
+## Emitted when [member character] got stuck while walking.
+signal got_stuck
+
+## Emitted when [member direction] is updated.
+signal direction_changed
+
+## The character walking speed.
+@export_range(10, 100000, 10, "or_greater", "suffix:m/s") var walk_speed: float = 300.0
+
+## The speed to consider that the character is stuck.
+## If less than [member walk_speed], the character may slide on walls instead of emitting
+## the [signal got_stuck] signal.
+## If closer to zero, the character may not ever emit the [signal got_stuck] signal.
+@export_range(0, 100000, 10, "or_greater", "suffix:m/s") var stuck_speed: float = 100.0
+
+## The turn direction will be randomly picked between this and [member turn_angle_end].
+@export_range(0, 180, 1, "radians_as_degrees") var turn_angle_left: float = PI / 2.0
+
+## The turn direction will be randomly picked between [member turn_angle_start] and this.
+@export_range(0, 180, 1, "radians_as_degrees") var turn_angle_right: float = PI / 2.0
```

Comment:
The docstrings refer to turn_angle_end and _start, while the properties are _left and _right. I think I gently prefer left and right.

(I would suggest we use _anticlockwise and _clockwise but not only are these really long words, I remembered that US English (which CONTRIBUTING says is the correct dialect) says _counterclockwise not anti. dextrorotatory and levorotatory? 😁 )